src/todo_collection.js

```diff
@@ -172,12 +170,11 @@ class ToDoCollection {
     this.render();
   }
 
-  delTask(index) {
-    this.todoData.splice(index, 1);
-    for (let i = index; i < this.todoData.length; i += 1) {
-      this.todoData[i].index -= 1;
-    }
-    localStorage.setItem('todoData', JSON.stringify(this.todoData));
+  delTask() {
+    const falseTask = this.todoData.filter((item) => item.completed === false);
+    for (let i = 0; i < falseTask.length; i += 1) falseTask[i].index = i;
+    localStorage.setItem('todoData', JSON.stringify(falseTask));
```

Comment:
Great job @lRebornsl 🤝

- [ ] As stated in the project requirements, kindly ensure the task object's values for `completed` is updated upon user actions.
![completed](https://github.com/lRebornsl/todo-list/assets/63917725/d2d30b21-24a4-4804-a404-f4202d318584)

Reply:
Solved